src/geometry/ArborX_Point.hpp

```diff
@@ -35,11 +37,33 @@ struct Point
   Coordinate _coords[DIM] = {};
 };
 
-template <typename... T>
-KOKKOS_FUNCTION Point(T...)
-    -> Point<sizeof...(T), std::conditional_t<
-                               (... || std::is_same_v<std::decay_t<T>, double>),
-                               double, float>>;
+// If all inputs are integral, convert to double. Otherwise, all inputs must be
+// the same type.
+// Note: converting integral types to double may introduce errors for 8+ byte
+// sized integers.k
+template <typename T, typename... Ts>
+#if KOKKOS_VERSION >= 40400
+KOKKOS_DEDUCTION_GUIDE
+#else
+KOKKOS_FUNCTION
+#endif
+Point(T, Ts...) -> Point<
```

Comment:
Why do we have both `T` and `Ts`?

Reply:
So that we could check that all types are identical. It feels easier than trying to get the first type from a parameter pack.

Reply:
I thought users may benefit from that one too. I don't have a strong need for it at the moment.

Reply:
You mean not needing brace initialization?

Reply:
I think users may expect both `Point(a, b, c)` and `Point({a, b, c})` working.